instance/models/mixins/openedx_database.py

```diff
@@ -234,6 +236,19 @@ def set_field_defaults(self):
         if not self.mongo_user:
             self.mongo_user = get_random_string(length=16, allowed_chars=string.ascii_lowercase)
             self.mongo_pass = get_random_string(length=32)
+        if not self.rabbitmq_vhost:
+            def new_rabbitmq_user():
```

Comment:
This method became more complicated, so I will make this a static method of `OpenEdXDatabaseMixin`.